clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/PartitionConsumptionState.java

```diff
@@ -59,7 +63,41 @@ public class PartitionConsumptionState {
   private boolean completionReported;
   private boolean isSubscribed;
   private boolean isDataRecoveryCompleted;
-  private LeaderFollowerStateType leaderFollowerState;
+
+  /**
+   * The lock is mainly to protect the consumer thread (while processing inflight messages) from the SIT thread trying
+   * to update the state during a state transition. The read lock is acquired in the consumer thread for the entirety
+   * of one message's processing in {@link StoreIngestionTask#produceToStoreBufferServiceOrKafka}, and the SIT must wait
+   * for the release of the read lock before it can acquire the write lock and update the state. That is the only
+   * lengthy acquisition of the lock.
+   */
+  final class LeaderFollowerState {
+    private LeaderFollowerStateType state;
+    private final ReadWriteLock rwLock;
+
+    LeaderFollowerState() {
+      this.state = LeaderFollowerStateType.STANDBY;
+      this.rwLock = new ReentrantReadWriteLock();
```

Comment:
Maybe here we should use the venice flavor of ReentrantReadWriteLock: `VeniceReentrantReadWriteLock` for debugging purpose. 

Besides, this lock was by default as unfair,  it does not prefer writing. But for our use-case, we may need to prefer write a bit as for heavy ingestion case we do not want to block state transition for long time.

Reply:
Omg that class it not used anywhere else in the codebase...

I moved the `VeniceReentrantReadWriteLock` from `venice-controller` to `venice-client-common` and used it in this file.

Reply:
Regarding the default as fair or unfair, what's your preference?